ui/components/NodeInputButton.tsx

```diff
@@ -3,6 +3,16 @@ import { Button } from "@canonical/react-components";
 import React, { FC, FormEvent } from "react";
 import { NodeInputProps } from "./helpers";
 
+const getWebAuthnRegistrationPayload = (evalCode: string): unknown => {
+  const tmp = evalCode.replace("window.__oryWebAuthnRegistration(", "");
+  const raw = tmp.substring(0, tmp.length - 1);
+  return JSON.parse(raw) as unknown;
+};
```

Comment:
question: sorry but I must ask: don't we have any other way to do this?

Reply:
If we want to avoid the eval, we need to have a custom parser. If you have an idea how to improve this?

Reply:
Custom parser would be out of the discussion, so I'd say we don't have another way of doing this